enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/controller/api/OnboardingController.java

```diff
@@ -183,4 +187,50 @@ public Response performCleanup(@EncryptedRequestBody ObjectRequest<OnboardingCle
         return onboardingService.performCleanup(request.getRequestObject());
     }
 
-}
\ No newline at end of file
+    @PostMapping("consent/text")
+    @Operation(
+            summary = "Obtain consent text",
+            description = "Obtain a text of user consent in specified language."
+    )
+    public ObjectResponse<OnboardingConsentTextResponse> fetchConsentText(final @RequestBody OnboardingConsentTextRequest request) throws OnboardingProcessException{
+        logger.debug("Returning consent for {}", request);
+        validate(request);
+        final OnboardingConsentTextResponse onboardingConsentTextResponse = onboardingService.fetchConsentText(request);
+        return new ObjectResponse<>(onboardingConsentTextResponse);
+    }
+
+    @PostMapping("consent/approve")
+    @Operation(
+            summary = "Store user consent",
+            description = "Store user consent, whether approved or not."
+    )
+    public Response approveConsent(final @RequestBody OnboardingConsentApprovalRequest request) throws OnboardingProcessException {
+        logger.debug("Approving consent for {}", request);
+        validate(request);
+        onboardingService.approveConsent(request);
+        return new Response();
+    }
+
+    private static void validate(final OnboardingConsentTextRequest request) {
+        if (StringUtils.isBlank(request.getUserId())
+                || StringUtils.isBlank(request.getLanguage())
+                || request.getConsentType() == null
+                || request.getProcessId() == null) {
+            throw new ValidationException("Missing mandatory attributes");
+        }
+        try {
+            LocaleUtils.toLocale(request.getLanguage());
+        } catch (IllegalArgumentException e) {
+            throw new ValidationException("Invalid language: " + request.getLanguage(), e);
+        }
+    }
+
+    private static void validate(final OnboardingConsentApprovalRequest request) {
```

Comment:
Or do you prefer to replace it by _javax.validation_?

Reply:
Typically, we would isolate the logic into a separate `OnboardingConsentApprovalRequestValidator` class.